presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoDatabaseMetaData.java

```diff
@@ -1247,29 +1238,41 @@ public int getResultSetHoldability()
     public int getDatabaseMajorVersion()
             throws SQLException
     {
-        // TODO: get version from server
-        return PrestoDriver.VERSION_MAJOR;
+        return getDatabaseVersionPart(0);
     }
 
     @Override
     public int getDatabaseMinorVersion()
             throws SQLException
     {
-        return PrestoDriver.VERSION_MINOR;
+        return getDatabaseVersionPart(1);
+    }
+
+    private int getDatabaseVersionPart(int part)
+            throws SQLException
+    {
+        String version = getDatabaseProductVersion();
+        List<String> parts = Splitter.on('.').limit(3).splitToList(version);
+        try {
+            return parseInt(parts.get(part));
+        }
+        catch (IndexOutOfBoundsException | NumberFormatException e) {
+            return 0;
```

Comment:
1. If the driver JAR is built from Maven, it will work correctly, since the snapshot JAR will have the correct manifest. I just tested this out (and fixed a bug). The version for snapshots is a bit weird -- it's the output of `git describe HEAD`. So, for this branch on `0.162-SNAPSHOT`, the version is `0.161-115-gea22394b5` (and that depends on having fetched the tags to the local Git repo). Thus, you get 0/161 for major/minor, and the full string for `getDriverVersion()`.
2. I considered that, but it will break for stuff like `presto.version=testversion` that we have in the development `config.properties` file. I'd rather be conservative for now.